test/files/run/t9529-types/Test_1.scala

```diff
@@ -0,0 +1,26 @@
+/* evidently annotations on types don't make it into bytecode yet, even though
+ * such a thing is allowed in Java 8 and onwards. Here's a test that it'll work
+ * with repeatable annotations anyways. */
+
+import scala.tools.partest._
+
+import anns._
+
+@TypeAnn_0("")
+object Test extends DirectTest {
+
+  override def extraSettings: String =
+    s"-usejavacp -cp ${testOutput.path} -Xprint:pic,ref -Ystop-after:ref -d ${testOutput.path}"
+
+  override def code =
+    """import anns._
+      |trait Foo extends (
+      |  (Int @TypeAnn_0("a") @TypeAnn_0("b"))
+      |    => (String @TypeAnn_0("x") @TypeAnn_0("y"))
+      |) {
+      |  type Meh = Any@TypeAnn_0("p")@TypeAnn_0("q")
```

Comment:
I had `val meh: Any @ ...`, but typer strips them due to not being `<: TypeConstraint`, so I'm not sure what to do. I wonder if Java annotations with `@Target(ElementType.TYPE_USE)` should be considered typely constraining. I wonder if anyone, anywhere, cares.

Reply:
That's fine, no action required (here) :-)